### ISSUE TITLE
grpc::node::Server::ShutdownServer() - Cannot create a handle without a HandleScope

### ISSUE THREAD

**Issue by gpean** [grpc/grpc#10557]
Seen with node v6.9.1 + grpc-1.2.3 at least 3 times, but seems pretty random - I won't be able to extract/post the workload that lead to this, sorry. Hopefully you can think of a simple fix for this. Looks race-condition-ish when garbage collection kicks in?

```
FATAL ERROR: v8::HandleScope::CreateHandle() Cannot create a handle without a HandleScope
 1: node::Abort() [node]
 2: 0x1098b2c [node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [node]
 4: v8::internal::HandleScope::Extend(v8::internal::Isolate*) [node]
 5: v8::EscapableHandleScope::EscapableHandleScope(v8::Isolate*) [node]
 6: grpc::node::Server::ShutdownServer() [/srv/dta-dmack/companion/node_modules/grpc/src/node/extension_binary/grpc_node.node]
 7: grpc::node::Server::~Server() [/srv/dta-dmack/companion/node_modules/grpc/src/node/extension_binary/grpc_node.node]
 8: grpc::node::Server::~Server() [/srv/dta-dmack/companion/node_modules/grpc/src/node/extension_binary/grpc_node.node]
 9: v8::internal::GlobalHandles::DispatchPendingPhantomCallbacks(bool) [node]
10: v8::internal::GlobalHandles::PostGarbageCollectionProcessing(v8::internal::GarbageCollector, v8::GCCallbackFlags) [node]
11: v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [node]
12: v8::internal::Heap::CollectGarbage(v8::internal::GarbageCollector, char const*, char const*, v8::GCCallbackFlags) [node]
13: v8::internal::Heap::FinalizeIncrementalMarkingIfComplete(char const*) [node]
14: v8::internal::IncrementalMarkingJob::DelayedTask::RunInternal() [node]
15: v8::platform::DefaultPlatform::PumpMessageLoop(v8::Isolate*) [node]
16: node::Start(int, char**) [node]
17: __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
18: 0x7b4f6d [node]
```

**Comment by jammi** [comment]
I'm getting this same error. Seems to be caused by just a certain amount of calls and then it fails. I encounter this in my tests.

**Comment by murgatroid99** [comment]
That's interesting. The stacktrace above shows that this specific failure happens when the Server gets garbage collected. That shouldn't be happening at all if the server is supposed to be handling calls.

**Comment by murgatroid99** [comment]
This should be fixed in gRPC 1.2.4
